docs/java-rest/high-level/migration.asciidoc

```diff
@@ -0,0 +1,137 @@
+
+Java application that uses the `TransportClient` declare the following dependency:
+
+["source","xml",subs="attributes"]
+--------------------------------------------------
+<dependency>
+    <groupId>org.elasticsearch.client</groupId>
+    <artifactId>transport</artifactId>
+    <version>...</version>
+</dependency>
+--------------------------------------------------
+
+This dependency must be removed and replaced by a new dependency on the
+high-level client. The <<java-rest-high-usage,Getting Started>> page shows
+ typical configurations for Maven and Gradle and presents the
+ <<java-rest-high-usage-dependencies, dependencies>> brought by the
+ high-level client.
+
+==== Changing the client's initialization code
+
+The `TransportClient` is typically initialized with something like:
+[source,java]
+--------------------------------------------------
+Settings settings = Settings.builder()
+        .put("cluster.name", "prod").build();
+
+TransportClient client = new PreBuiltTransportClient(settings)
+        .addTransportAddress(new TransportAddress(InetAddress.getByName("host"), 9300));
+--------------------------------------------------
+
+The initialization of a `RestHighLevelClient` is different. It first requires the initialization
+of a <<java-rest-low-usage-initialization,low-level client>>:
+
+[source,java]
+--------------------------------------------------
+RestClient lowLevelRestClient = RestClient.builder(
+                new HttpHost("host", 9200, "http")).build();
+--------------------------------------------------
+
+Which is then passed to the constructor of the `RestHighLevelClient`:
+
+[source,java]
+--------------------------------------------------
+RestHighLevelClient client =
+    new RestHighLevelClient(lowLevelRestClient);
+--------------------------------------------------
+
+Both `RestClient` and `RestHighLevelClient` are thread safe. They are
+ typically instantiated by the application at startup time or when the
+ first request is executed, and must both be closed when they are not
+ needed anymore or when the application is stopped.
+
+The `RestClient` provides various <<java-rest-low-usage-initialization,
+ initialization options>> that can be used to tune the way it executes
+ the requests.
+
+Once the `RestHighLevelClient` is initialized, it can then be used to
+execute any of the <<java-rest-high-supported-apis,supported APIs>>.
+
+WARNING: The high-level client does not verify the name of the cluster
+it connects to. This verification can be made after the initialization
+of the client by checking the <<java-rest-high-main, cluster's information>>
+
+==== Changing the application's code
+
+The `RestHighLevelClient` does not provide the same methods than
+the `TransportClient` to execute the requests.
+
+More important, the high-level client:
+
+- does not provide indices or cluster management APIs. Management
+operations must be executed by external scripts or using the low-level
+client.
```

Comment:
does not provide (yet)  ? we may provide them at some point... would be nice to add an example of using the low-level client for some simple admin operation.